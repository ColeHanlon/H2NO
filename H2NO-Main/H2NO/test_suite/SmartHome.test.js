import React from 'react';
import renderer from 'react-test-renderer';
import SmartHomePage from '../components/smart_home_page';
import { TouchableOpacity } from 'react-native';
// import {shallow} from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';

describe('<Smart Home/>', () => {
  it("has 8 children", () => {
    const tree = renderer.create(<SmartHomePage/>).toJSON();
    expect(tree.children.length).toBe(8);
  })
  it('renders correctly', () => {
    const tree = renderer.create(<SmartHomePage/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
  // it('click Thermostat', () => {
  //   const tree = renderer.create(<SmartHomePage/>);
  //   const mockcallback = jest.fn();
  //   const button = shallow((<TouchableOpacity onPress={mockcallback}>ok!</TouchableOpacity>));
  //   button.find('button').simulate('click');
  //   expect(mockCallBack.mock.calls.length).toEqual(1);
  // })
})