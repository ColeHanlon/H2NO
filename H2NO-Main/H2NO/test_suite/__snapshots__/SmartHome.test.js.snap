// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Smart Home/> renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "color": "#000",
      "fontSize": "35px",
      "left": 175,
      "position": "relative",
      "top": 300,
    }
  }
>
  <Text
    style={
      Object {
        "alignContent": "center",
        "fontSize": "30px",
        "left": -60,
        "position": "absolute",
        "top": -250,
      }
    }
  >
    Smart Home
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#00d",
        "borderRadius": 10,
        "elevation": 8,
        "left": -150,
        "opacity": 1,
        "paddingHorizontal": 17,
        "paddingVertical": 15,
        "position": "absolute",
        "top": -175,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      thermostat
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#00d",
        "borderRadius": 10,
        "elevation": 8,
        "left": 50,
        "opacity": 1,
        "paddingHorizontal": 17,
        "paddingVertical": 15,
        "position": "absolute",
        "top": -175,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      fans
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#00d",
        "borderRadius": 10,
        "elevation": 8,
        "left": -150,
        "opacity": 1,
        "paddingHorizontal": 17,
        "paddingVertical": 15,
        "position": "absolute",
        "top": -75,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      sprinklers
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#00d",
        "borderRadius": 10,
        "elevation": 8,
        "left": 50,
        "opacity": 1,
        "paddingHorizontal": 17,
        "paddingVertical": 15,
        "position": "absolute",
        "top": -75,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      heater
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#00d",
        "borderRadius": 10,
        "elevation": 8,
        "left": -150,
        "opacity": 1,
        "paddingHorizontal": 17,
        "paddingVertical": 15,
        "position": "absolute",
        "top": 25,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      blinds
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#00d",
        "borderRadius": 10,
        "elevation": 8,
        "left": 0,
        "opacity": 1,
        "paddingHorizontal": 17,
        "paddingVertical": 15,
        "position": "absolute",
        "top": 25,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      aircooler
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#00d",
        "borderRadius": 10,
        "elevation": 8,
        "left": -150,
        "opacity": 1,
        "paddingHorizontal": 17,
        "paddingVertical": 15,
        "position": "absolute",
        "top": 125,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      lights
    </Text>
  </View>
</View>
`;
